### DISCUSSION TITLE
Getting a list of all available units

### DISCUSSION THREAD

**Discussion by janniklas-jakob** [josdejong/mathjs#2805] [Q&A] [answered]
Hi all, 

I'm using mathjs mainly for its unit functionality. Specifically, users can input values and units to convert between them. Later, I might add calculations.

It would be helpful to give the users a list of all units that are available (even user-defined ones). I've been looking at the documentation and only found a list in the [unit references](https://mathjs.org/docs/datatypes/units.html#reference). 

Am I over-looking something? Is this functionality already available? If not, this is a feature request ;)

**Comment by gwhitney** [comment, marked as answer]
I believe the list you are looking for is in math.Unit.UNITS.  Give that a try and let us know if it has the information you need.

**Reply by janniklas-jakob** [reply]
From the [source code](https://github.com/josdejong/mathjs/blob/5754478f168b67e9774d4dfbb5c4f45ad34f97ca/src/type/unit/Unit.js#L1528) it does look like the correct object I was looking for. Thank you! I just can't seem to access it in my typescript project...

```
import { all, create } from "mathjs";
const math = create(all);
console.log(math.Unit.UNITS);
```
I get `Property 'Unit' does not exist on type 'MathJsStatic'. Did you mean 'unit'? ts(2551)`. However, in the debugger I can see the UNITS object.
Could this be a Typescript issue?

**Reply by gwhitney** [reply]
Hmm, seems quite possible. Maybe do a test where you tell typescript to ignore that error and see if you get the results you want? If so, then file an issue that math.Unit is not exposed in Typescript. Thanks.

**Reply by josdejong** [reply]
Just for reference, here is the issue reporting these missing type definitions: #2286
